src/serac/physics/thermal_conduction_functional.hpp

```diff
@@ -153,10 +153,12 @@ class ThermalConductionFunctional : public BasePhysics {
 
     functional_call_args_.emplace_back(temperature_.trueVec());
 
-    for (size_t i = 0; i < sizeof...(parameter_space); ++i) {
-      trial_spaces[i + 1]         = &(parameter_states_[i].get().space());
-      parameter_sensitivities_[i] = std::make_unique<FiniteElementDual>(mesh_, parameter_states_[i].get().space());
-      functional_call_args_.emplace_back(parameter_states_[i].get().trueVec());
+    if constexpr (sizeof...(parameter_space)) {
```

Comment:
Why is this needed?

Reply:
oops, I was a getting compiler error inside the `for` loop (just below the line of code you highlighted):

```cpp
for (size_t i = 0; i < sizeof...(parameter_space); ++i) {
                                      ^
```

When `parameter_space` is empty, then the end condition on the for loop is essentially just the literal `0`.
The compiler was complaining the comparison `i < 0` made the loop trivial.